internal/cluster/distribution/eks/node_pool.go

```diff
@@ -45,13 +45,21 @@ func (n NewNodePool) Validate() error {
 	var violations []string
 
 	if n.Autoscaling.Enabled {
-		if n.Autoscaling.MinSize < 1 {
-			violations = append(violations, "minimum autoscaling size cannot be lower than one")
+		if n.Autoscaling.MinSize < 0 {
+			violations = append(violations, "minimum autoscaling size cannot be negative")
 		}
 
 		if n.Autoscaling.MaxSize <= n.Autoscaling.MinSize {
```

Comment:
**Note**: kind of out of scope, but while we are here: this condition disallows creating fixed size (`minSize == maxSize`) autoscaled node pools. It's worth checking whether this is deliberate. As long as we don't allow changing node pool size constraints it is OK IMO, but when that design changes this needs to be revised.
Also the current wording of the error doesn't include the case I just talked about which might be confusing for the reader.
(It surely confused me whether the condition or the message reflects the true intent and whichever makes the other one incorrect.)

Reply:
AFAIK the min and max size cannot be the same, the max size needs to be at least larger than the min by one.

Note that we don't actually use the autoscaling capabilities of ASGs, so this is purely an upper limit for the cluster autoscaler. Fixed size node pool in our case means an ASG with a fix desired size, disabled autoscaling policies, but the ASG still requires minimum and maximum sizes.

Reply:
I see, thanks for the clarification.

Reply:
if autoscaling is not enabled, we don't use the min/max values from this struct, the asg is created with min=size and max=size+1 instead. so if you don't want autoscaling, you should set enabled=false and you can leave the min/max sizes zero. i don't know about any technical constraint from aws because of which we'd have to ensure that max is more than min, but in case of enabled autoscaling it makes sense to require this here